pkg/operator/operator.go

```diff
@@ -92,10 +92,17 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
 
 	// Start controllers
 	klog.Info("Starting controller-manager")
-	go o.manager.Start(ctx.Done())
+	go func() {
+		err := o.manager.Start(ctx.Done())
+		if err != nil {
+			klog.Fatalf("Failed to start controller-runtime manager: %v", err)
+		}
+	}()
 	go logLevelController.Run(ctx, 1)
 	go managementStateController.Run(ctx, 1)
-	connectivitycheck.Start(ctx, o.ccfg.KubeConfig)
+	if err := connectivitycheck.Start(ctx, o.ccfg.KubeConfig); err != nil {
+		klog.Errorf("Failed to start connectivitycheck controller: %v", err)
```

Comment:
Would it make sense to return here? Otherwise we just log and leave the operator up half functioning

Reply:
the connectivity check controller is not critical functionality; if it fails to start up (which is very unlikely here) we still want to try and continue to progress the cluster.

Reply:
Understood, thanks